### PR TITLE
Editor components

### PR DESCRIPTION
**Pull request by jossmac**
First pass at editor components. New package "@voussoir/editor" exports:

- `EditorPopover`
- `EditorToolbar`
- `EditorToolbarButton`
- `EditorToolbarGroup`
- `EditorToolbarItem`
- `EditorToolbarSeparator`

**Related changes:**

New `ToggleButton` exported from "@voussoir/button" package.

### CODE REVIEW

File: design-system/packages/editor/src/EditorToolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        e.preventDefault();
+        e.stopPropagation();
+        if (e.key === 'ArrowLeft' && isRtl) {
+          focusManager.focusNext({ wrap: true });
+        } else {
+          focusManager.focusPrevious({ wrap: true });
+        }
+        break;
```
Comment (emmatown) [inline design-system/packages/editor/src/EditorToolbar.tsx:371]: just a random idea: what if we supported the same keys as word navigation (option+left/right arrows on mac and i think ctrl+shift+left/right arrows on windows) to go between groups

Comment (jossmac) [reply design-system/packages/editor/src/EditorToolbar.tsx:371]: yeah, that could be nice. i'm not sure how to implement w/o skipping items that are outside of a group though 🤔

Comment (emmatown) [reply design-system/packages/editor/src/EditorToolbar.tsx]: i'd probably think of it as items outside of a group are essentially their own group? (just like words can be single characters)

Comment (jossmac) [reply design-system/packages/editor/src/EditorToolbar.tsx]: yeah, agreed. more of a technical concern; just need to think about how to filter the nodes to get the desired result

**Review by emmatown — APPROVED** [review]

**Comment by jossmac** [comment]
Replaced by #290

**Pull request closed without merging**